components/brave_wallet/browser/brave_wallet_utils.cc

```diff
@@ -948,4 +949,18 @@ absl::optional<std::string> GetPrefKeyForCoinType(mojom::CoinType coin) {
   return absl::nullopt;
 }
 
+bool ParseDataURIAndExtractJSON(const GURL url, std::string* json) {
+  std::string mime_type, charset, data;
+  if (!net::DataURL::Parse(url, &mime_type, &charset, &data) || data.empty()) {
+    return false;
+  }
+
+  if (mime_type != "application/json") {
+    return false;
+  }
+
+  *json = data;
+  return true;
+}
```

Comment:
I wasn't sure where to put this function.

Reply:
might makes more sense in eth_response_parser?

Reply:
That's probably right.  I wasn't sure because don't think this logic is eth specific (there is no Ethereum specific decoding, and in theory Solana NFTs could also be require similar parsing).

Reply:
Updated in 3957076e42002b574e83fc793c171046cc43f5e8